File: book/petrol-pump/java/pump/Inputs.java
package pump;

import nz.sodium.*;

public class Inputs {
    public Inputs(
            Stream<UpDown> sNozzle1,
            Stream<UpDown> sNozzle2,
            Stream<UpDown> sNozzle3,
            Stream<Key> sKeypad,
            Stream<Integer> sFuelPulses,
            Cell<Double> calibration,
            Cell<Double> price1,
            Cell<Double> price2,
            Cell<Double> price3,
            Stream<Unit> sClearSale) {
        this.sNozzle1 = sNozzle1;
        this.sNozzle2 = sNozzle2;
        this.sNozzle3 = sNozzle3;
        this.sKeypad = sKeypad;
        this.sFuelPulses = sFuelPulses;
        this.calibration = calibration;
        this.price1 = price1;
        this.price2 = price2;
        this.price3 = price3;
        this.sClearSale = sClearSale;
    }

    public final Stream<UpDown> sNozzle1;
    public final Stream<UpDown> sNozzle2;
    public final Stream<UpDown> sNozzle3;
    public final Stream<Key> sKeypad;
    public final Stream<Integer> sFuelPulses;
    public final Cell<Double> calibration;
    public final Cell<Double> price1;
    public final Cell<Double> price2;
    public final Cell<Double> price3;
    public final Stream<Unit> sClearSale;
}

File: book/petrol-pump/c#/PetrolPump/Inputs.cs
using Sodium.Frp;
using Sodium.Functional;

namespace PetrolPump
{
    public class Inputs
    {
        public Inputs(
            Stream<UpDown> sNozzle1,
            Stream<UpDown> sNozzle2,
            Stream<UpDown> sNozzle3,
            Stream<Key> sKeypad,
            Stream<int> sFuelPulses,
            Cell<double> calibration,
            Cell<double> price1,
            Cell<double> price2,
            Cell<double> price3,
            Stream<Unit> sClearSale)
        {
            this.SNozzle1 = sNozzle1;
            this.SNozzle2 = sNozzle2;
            this.SNozzle3 = sNozzle3;
            this.SKeypad = sKeypad;
            this.SFuelPulses = sFuelPulses;
            this.Calibration = calibration;
            this.Price1 = price1;
            this.Price2 = price2;
            this.Price3 = price3;
            this.SClearSale = sClearSale;
        }

        public Stream<UpDown> SNozzle1 { get; }
        public Stream<UpDown> SNozzle2 { get; }
        public Stream<UpDown> SNozzle3 { get; }
        public Stream<Key> SKeypad { get; }
        public Stream<int> SFuelPulses { get; }
        public Cell<double> Calibration { get; }
        public Cell<double> Price1 { get; }
        public Cell<double> Price2 { get; }
        public Cell<double> Price3 { get; }
        public Stream<Unit> SClearSale { get; }
    }
}